llvm/include/llvm/IR/IRBuilder.h

```diff
@@ -87,6 +87,20 @@ class IRBuilderCallbackInserter : public IRBuilderDefaultInserter {
   }
 };
 
+/// This provides a helper for copying FMF from an instruction or setting
+/// specified flags.
+struct FMFSource final {
```

Comment:
```suggestion
struct FMFSource {
```
Don't need final for non-virtual classes.